backend/test/forgotPassword.test.js

```diff
@@ -0,0 +1,125 @@
+      username: 'johnwick123',
+    };
+
+    chai
+      .request(server)
+      .post('/admin/superAdmin')
+      .send(adminData)
+      .end((err, res) => {
+        expect(err).to.equal(null);
+        expect(res.status).to.equal(200);
+        expect(res.body.firstName).to.equal(adminData.firstName);
+        expect(res.body.lastName).to.equal(adminData.lastName);
+        expect(res.body.isSuperAdmin).to.equal(true);
+        done();
+      });
+  });
+
+  // Step 2 - We log in the admin using the above credentials
+  it('Logging in with the old password', (done) => {
+    const adminData = {
+      email: 'john@wick.com',
+      password: '123asd',
+    };
+
+    chai
+      .request(server)
+      .post('/auth/login')
+      .send(adminData)
+      .end((err, res) => {
+        expect(err).to.equal(null);
+        expect(res.status).to.equal(200);
+        expect(res.body.name).to.equal('john wick');
+        expect(res.body.email).to.equal(adminData.email);
+        expect(res.body.isSuperAdmin).to.equal(true);
+        expect(res.body.token).to.be.a('string');
+        done();
+      });
+  });
+
+  // Step 3 - We send a request to /forgotpassword and get a URL back
+  it('Sending a forgot password request', (done) => {
+    const adminData = {
+      email: 'john@wick.com',
+    };
+
+    chai
+      .request(server)
+      .post('/admin/forgotpassword')
+      .send(adminData)
+      .end((err, res) => {
+        forgotPasswordURL = res.body.resetPasswordURL.replace('http://localhost:3500', ''); // Add the path with token to this varible for our next test case
+        expect(err).to.equal(null);
+        expect(res.status).to.equal(200);
+        expect(res.body.resetPasswordURL).to.include('http://localhost:3500/admin/forgotpassword/');
+        done();
+      });
+  });
+
+  // Step 4 - Chaning our password by sending the request to the above URL (forgotPasswordURL)
+  it('Resetting new password', (done) => {
+    const adminData = {
+      newPassword: 'newpassword123',
+    };
+
+    chai
+      .request(server)
+      .post(forgotPasswordURL)
+      .send(adminData)
+      .end((err, res) => {
+        expect(err).to.equal(null);
+        expect(res.status).to.equal(200);
+        expect(res.body.message).to.equal('Reset Password Successful');
+        done();
+      });
+  });
+
+  // Step 5 - Login admin with the new password
+  it('Logging in with the new password', (done) => {
+    const adminData = {
+      email: 'john@wick.com',
+      password: 'newpassword123',
+    };
+
+    chai
+      .request(server)
+      .post('/auth/login')
+      .send(adminData)
+      .end((err, res) => {
+        expect(err).to.equal(null);
+        expect(res.status).to.equal(200);
+        expect(res.body.name).to.equal('john wick');
+        expect(res.body.email).to.equal(adminData.email);
+        expect(res.body.isSuperAdmin).to.equal(true);
+        expect(res.body.token).to.be.a('string');
+        done();
+      });
+  });
```

Comment:
Can you figure out a way to end the process after all the tests are run?

Reply:
End what process? Can you please elaborate?

Reply:
the node server, one needs to manually hit ctrl + c to stop the server after the test is complete

Reply:
Added `--exit` flag while running mocha which is automatically terminate the process after running the test cases